custom_components/thermal_comfort/config_flow.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+import voluptuous as vol
+import logging
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_NAME,
+)
+from homeassistant.core import callback
+
+from .const import (
+    CONF_TEMPERATURE_SENSOR,
+    CONF_HUMIDITY_SENSOR,
+    CONF_POLL,
+    DEFAULT_NAME,
+    DOMAIN,
+)
+from .sensor import SensorType
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_value(config_entry: config_entries.ConfigEntry | None, param: str, default=None):
+    """Get current value for configuration parameter
+
+    :param config_entry: config_entries|None: config entry from Flow
+    :param param: str: parameter name for getting value
+    :param default: default value for parameter, defaults to None
+    :returns: parameter value, or default value or None
+    """
+    if config_entry is not None:
+        return config_entry.options.get(param, config_entry.data.get(param, default))
+    else:
+        return default
+
+
+def build_schema(config_entry: config_entries | None) -> vol.Schema:
+    # ToDo: get list of CONF_TEMPERATURE_SENSOR and CONF_HUMIDITY_SENSOR to create dropdown list and "one of" selection
+    schema = vol.Schema({
+        vol.Required(CONF_NAME, default=get_value(config_entry, CONF_NAME, DEFAULT_NAME)): str,
+        vol.Required(CONF_TEMPERATURE_SENSOR, default=get_value(config_entry, CONF_TEMPERATURE_SENSOR)): str,
+        vol.Required(CONF_HUMIDITY_SENSOR, default=get_value(config_entry, CONF_HUMIDITY_SENSOR)): str,
+        vol.Optional(CONF_POLL, default=get_value(config_entry, CONF_POLL, False)): bool,
+    })
+    for st in SensorType:
+
+        schema = schema.extend({
+            vol.Optional(str(st), default=get_value(config_entry, str(st), True)): bool
+        })
+
+    return schema
+
+
+class ThermalComfortConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return ThermalComfortOptionsFlow(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if user_input is not None:
+            await self.async_set_unique_id(f"{user_input[CONF_TEMPERATURE_SENSOR]}-{user_input[CONF_HUMIDITY_SENSOR]}")
+            self._abort_if_unique_id_configured()
+
+            # ToDo: we should not trust user and check:
+            #  - that user_input[CONF_TEMPERATURE_SENSOR] is temperature sensor
+            #  - that user_input[CONF_HUMIDITY_SENSOR] is humidity sensor
+
```

Comment:
We probably should move poll to advanced option.

We should also discuss if same should happen with sensor selection. I can't find the source right now, but if i remember correctly the general idea is to not prevent sensor creation in config_flow but disable the unneeded sensors through the frontend.